lib/puppet/provider/base_dsc/powershell.rb

```diff
@@ -15,6 +17,49 @@
 Applies DSC Resources by generating a configuration file and applying it.
 EOT
 
+  def initialize(value={})
+    super(value)
+    set_symlink_folder
+  end
+
+  if Puppet::Util::Platform.windows?
```

Comment:
As far as order of operations here... 

I think we want the symlink creation only once when the provider is first created, not each time a new provider instance is associated with a path though, right?  Which I think means we want this in `self.prefetch` instead of in the initializer.

I'm testing this locally now (and will make sure I understand the testing implications, given I know you worked hard on getting all that setup).